typed-racket-lib/typed-racket/types/subtype.rkt

```diff
@@ -682,6 +671,16 @@
                  #:break (not A))
         (subtype* A elem1 seq-t))]
      [_ (continue<: A t1 t2 obj)])]
+  [(case: Immutable-HashTable (Immutable-HashTable: key1 val1))
+   (match t2
+    [(Immutable-HashTable: key2 val2)
+     (subtype-seq A (subtype* key1 key2) (subtype* val1 val2))]
```

Comment:
the rest of this file has a vertically-aligned style for the `subtype-seq` monad macro---might be nice to keep that style consistent... maybe it doesn't matter. I always associated the vertical style with it being like a `do` block a-lá Haskell.